### PR TITLE
Simple gradient descent optimiser

### LINKED ISSUE #930: Add simple gradient descent optimiser
See also #893 

Only for models with sensitivities

### PR DESCRIPTION
**Pull request by MichaelClerx**
Closes #930 

- Updates the optimiser interface to allow sensitivities
- Updates the optimiser controller to work with sensitivities
- Adds a simple fixed learning rate gradient descent method

### CODE REVIEW

**Comment by MichaelClerx** [comment]
@ben18785 & @alisterde could you have a quick look at this PR for me?
It adds in a first gradient based optimiser, which I think would pave the way for your BFGS optimiser.

**Comment by MichaelClerx** [comment]
Thanks @alisterde !

File: pints/_optimisers/_gradient_descent.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        """ See :meth:`Optimiser.fbest()`. """
+        return self._fbest
+
+    def learning_rate(self):
+        """ Returns this optimiser's learning rate. """
+        return self._eta
+
+    def name(self):
+        """ See :meth:`Optimiser.name()`. """
+        return 'Fixed learning rate gradient descent'
```
Comment (ben18785) [inline pints/_optimisers/_gradient_descent.py:62]: I would say just return 'Gradient descent' since this matches the class name...

**Comment by MichaelClerx** [comment]
Thanks @ben18785 ! I quite like how the notebook suggests improvements to this method :D

**Comment by MichaelClerx** [comment]
Have added some tests and changed the name now @ben18785 !

**Comment by MichaelClerx** [comment]
> This all looks good to me code-wise.
> 
> Is this the canonical "gradient descent" optimiser or, for instance, are there examples with non-constant learning rate? I.e. is the name going to need to become more specific at some point in the future when someone implements a similar algorithm?

It's a good question, we'll probably want to name it something more specific in the future. (Gradient descent is one of those things that became a class of methods, I suppose)

But to be honest I'm not super worried about that, because this won't really be used in practice

**Comment by MichaelClerx** [comment]
Then again, @fcooper8472 , the more specific ones do seem to have specific names, so could be fine to leave this as is even when we add more?

See https://github.com/pints-team/pints/issues/1105

Someone will probably do these for fun at some point :D

**Comment by fcooper8472** [comment]
> Then again, @fcooper8472 , the more specific ones do seem to have specific names, so could be fine to leave this as is even when we add more?

Sounds good to me.
